lrlex/src/main.rs

```diff
@@ -42,43 +47,71 @@ fn read_file(path: &str) -> String {
     s
 }
 
-fn main() {
+fn main() -> Result<(), Box<dyn Error>> {
     let args: Vec<String> = env::args().collect();
     let prog = args[0].clone();
     let matches = match Options::new().optflag("h", "help", "").parse(&args[1..]) {
         Ok(m) => m,
         Err(f) => {
             usage(&prog, f.to_string().as_str());
-            return;
+            return Ok(());
         }
     };
     if matches.opt_present("h") || matches.free.len() != 2 {
         usage(&prog, "");
-        return;
+        return Ok(());
     }
 
     let lex_l_path = &matches.free[0];
     let lex_src = read_file(lex_l_path);
-    let lexerdef = LRNonStreamingLexerDef::<DefaultLexerTypes<u32>>::from_str(&lex_src)
-        .unwrap_or_else(|errs| {
-            let nlcache = NewlineCache::from_str(&lex_src).unwrap();
-            for e in errs {
-                if let Some((line, column)) = nlcache
-                    .byte_to_line_num_and_col_num(&lex_src, e.spans().first().unwrap().start())
-                {
-                    writeln!(
-                        stderr(),
-                        "{}: {} at line {line} column {column}",
-                        &lex_l_path,
-                        &e
-                    )
-                    .ok();
-                } else {
-                    writeln!(stderr(), "{}: {}", &lex_l_path, &e).ok();
-                }
+    let lex_diag = SpannedDiagnosticFormatter::new(&lex_src, Path::new(lex_l_path));
+    let (mut header, _) = match GrmtoolsSectionParser::new(&lex_src, false).parse() {
+        Ok(x) => x,
+        Err(es) => {
+            eprintln!(
+                "\n{ERROR}{}",
+                lex_diag.file_location_msg(" parsing the `%grmtools` section", None)
+            );
+            for e in es {
+                eprintln!(
+                    "{}",
+                    &indent("     ", &lex_diag.format_error(e).to_string())
+                );
             }
             process::exit(1);
-        });
+        }
+    };
+    header.mark_used(&"lexerkind".to_string());
```

Comment:
Marking as used here, but never doing the check for unused values.

Reply:
Should be fixed in [51b4b97](https://github.com/softdevteam/grmtools/pull/580/commits/51b4b9727dbc8c60a81f5a4cd38977d37046349a)